### ISSUE TITLE
AWSMobileClient iOS Extension Compatibility

### ISSUE THREAD

**Issue by james-ff** [aws-amplify/aws-sdk-ios#2958]
**Describe the bug**
The AWSMobileClient library uses `UIApplication.shared` which is not available in a share extension context.
 
Currently we are forced to work with the AWS iOS SDK very manually in order to fetch the latest access token in our share extension.
 
There are workarounds for keyWindow access adopted by other libraries, notably PanModal which is an open source swift library from Slack https://github.com/slackhq/PanModal/pull/81/files.
 
Solving this issue would enable us to use exclusively Amplify / AWSMobileClient throughout our app and extension which would be ideal. 

**To Reproduce**
1. pod install AWSMobileClient for an extension target (share extension)
2. build / run the extension in Xcode
3. See error due to unavailable apis in an extension context

**Areas of the SDK you are using (AWSMobileClient, Cognito, Pinpoint, IoT, etc)?**
Want to use:
`pod "AWSMobileClient", "2.15.1"`
Having to use:
```
pod "AWSCore", "2.15.1"
pod "AWSCognitoIdentityProvider", "2.15.1"
pod "AWSCognitoIdentityProviderASF", "1.0.1"
```

**Environment(please complete the following information):**
 - SDK Version: 2.15.1
 - Dependency Manager: Cocoapods
 - Swift Version : 5
 - Xcode Version: 11.6

**Comment by DongQuanRui** [comment]
Related to this one? https://github.com/aws-amplify/aws-sdk-ios/issues/1522

**Comment by james-ff** [comment]
Yes, here I've included additional context for the use case I'm working with: having to use lower level libraries in the meantime to workaround the issue. Ideally only one library would be linked between the app and extension target
